src/figuresClasses.ts

```diff
@@ -1,11 +1,102 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error("Sides can't form a triangle");
+    }
+    this.shape = Shape.Triangle;
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
 
-export class Triangle implements Figure {}
+    return Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  radius: number;
+
+  constructor(color: Color, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+    this.shape = Shape.Circle;
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * Math.pow(this.radius, 2) * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Rectangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  width: number;
+
+  height: number;
+
+  constructor(color: Color, width: number, height: number) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+    this.shape = Shape.Rectangle;
+    this.color = color;
+    this.width = width;
+    this.height = height;
```

Comment:
[CODE KNOWLEDGE] - The Rectangle class constructor is also a candidate for refactoring to use TypeScript's parameter properties to avoid redundant code.